src/clr-core/styles/tokens/tokens-build.js

```diff
@@ -9,32 +9,52 @@
 const fs = require('fs');
 const theo = require('theo');
 
-function getRemValue(prop, base = 20) {
+const relativeValueStore = {};
+
+function getId() {
+  return Math.random()
+    .toString(36)
+    .substr(2, 9);
+}
+
+function getRemValue(prop, base) {
   return `${(parseInt(prop.value.slice(0, -2)) / base).toFixed(base === 20 ? 2 : 4)}rem`;
 }
 
 function isRelativePxValue(prop) {
   return prop.type === 'px' && !prop['absolute-value'];
 }
 
-function getPropValue(prop) {
-  return isRelativePxValue(prop) ? getRemValue(prop) : prop.value;
+function getPropValue(prop, base = 20) {
+  return isRelativePxValue(prop) ? getRemValue(prop, base) : prop.value;
+}
+
+function getPropString(prop, base = 20) {
+  // if the token is private but a relative value we generate a unique css custom prop
+  // to be able dynamically adjust it for apps using base 16px font size
```

Comment:
Why are we focused specifically on base 16px as an alternative?

Reply:
This func in particular generates a unique id for relative size private tokens so we can still scale rem values easily but also discourage apps from using them since they are a random string value. So if the app is already using base 16 then they simply add one line change and the relative values auto scale properly for them without much work.

Example of generated code

```scss
// tokens/index.scss 
// Non relative private values are static and dont create a public css prop
$cds-token-layout-grid-cols: 12;

// relative private values create a private css prop to allow easy change of scale if neeeded
$cds-token-typography-font-size-2: var(--cds_alqnkoi63, 0.60rem);  // 12px base 20

// tokens/_public.scss
// consumers discouraged from using private vars but still can easily use base 16 if migrating to Clarity by applying the attribute
[cds-base-font="16"] {
  --cds_alqnkoi63: 0.7500rem; // 12px base 16
}
```